components/dashboard/src/app/AppBlockingFlows.tsx

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { FC, lazy } from "react";
+import { User } from "@gitpod/gitpod-protocol";
+import { useCheckDedicatedSetup } from "../hooks/use-check-dedicated-setup";
+import { useFeatureFlags } from "../contexts/FeatureFlagContext";
+import { useQueryParams } from "../hooks/use-query-param";
+import { useCurrentUser } from "../user-context";
+import { FORCE_ONBOARDING_PARAM, FORCE_ONBOARDING_PARAM_VALUE } from "../onboarding/UserOnboarding";
+
+const UserOnboarding = lazy(() => import(/* webpackPrefetch: true */ "../onboarding/UserOnboarding"));
+const DedicatedOnboarding = lazy(() => import(/* webpackPrefetch: true */ "../dedicated-setup/DedicatedSetup"));
+
+// This component handles any flows that should come after we've loaded the user/orgs, but before we render the normal app chrome.
+// Since this runs before the app is rendered, we should avoid adding any lengthy async calls that would delay the app from loading.
+export const AppBlockingFlows: FC = ({ children }) => {
+    const user = useCurrentUser();
+    const checkDedicatedOnboaring = useCheckDedicatedSetup();
+    const { newSignupFlow } = useFeatureFlags();
+    const search = useQueryParams();
+
+    // This shouldn't happen, but if it does don't render anything yet
+    if (!user) {
+        return <></>;
+    }
+
+    // Handle dedicated onboarding if necessary
+    if (!checkDedicatedOnboaring.isLoading && checkDedicatedOnboaring.needsOnboarding) {
+        return <DedicatedOnboarding />;
```

Comment:
*double-checking*: this is only rendered if the `dedicated-setup` query param is `true`, or the `enableDedicatedOnboardingFlow` feature flag is set.

👍🏻

Reply:
Yah, if you haven't set the query param, or the feature flag, `checkDedicatedOnboaring.needsOnboarding` will be `false`

Reply:
Update: Decided to only show this if the query param is set for now. This lets us push forward w/ dev while we work through adding a check on the backend and incorporate that into the client when it's ready.